gui

```diff
@@ -320,67 +111,72 @@ guimode="$(cat "${DIRECTORY}/data/settings/App List Style")"
 #In Xlunch mode, the window must be centered, but xlunch only handles absolute offsets.
 get_positions() {
   #determine screen_width and screen_height
-  screen_dimensions="$(xrandr --nograb --current | awk -F 'connected |\\+|\\('  '/ connected.*[0-9]+x[0-9]+\+/ && $2 {printf $2 ", "}' | sed -n -e 's/^.*primary //p' | tr 'x+' ' ' | tr ',+' ' ')"
-  if [ -z "$screen_dimensions" ];then
-    # if screen_dimensions is empty, this could be a single monitor wayland display which does not have the word "primary" in the output
-    # workaround is to get the first output returned for the connected display
-    screen_dimensions="$(xrandr --nograb --current | awk -F 'connected |\\+|\\('  '/ connected.*[0-9]+x[0-9]+\+/ && $2 {printf $2 ", "}' | tr 'x+' ' ' | tr ',+' ' ')"
+  if [ -z "$screen_width" ] || [ -z "$screen_height" ];then
```

Comment:
I believe this will cause the UI to appear incorrectly if the screen resolution changes after pi-apps is first opened or if the user moves the GUI to another monitor. This is because the screen_width and screen_height variables will be set after the first call of get_positions and will not be updated again. This is a very fast function to run and there isn't any measureable benefit to not running dimensions/width/height every time

both of these were not issues

Reply:
Screen_width is already only really used for the initial launch to choose one of two window sizing options.
Before I made these changes, if the screen resolution changed after launch, the main yad window would not receive the sizing update but the details window would now appear in "small mode". That is a situation we have not really planned for, but nobody has reported such a situation, so I doubt it happens often.

If we really wanted to account for changing screen resolutions, it would take more edits than reverting this commit. Either the app list would need to be closed and relaunched with new dimensions, or something would need to resize it on-the-fly.